app/supplejack/extraction/connection.rb

```diff
@@ -0,0 +1,29 @@
+module Extraction
+  class Connection
```

Comment:
I don't fully understand what the purpose of separating Connection and Request are? Maybe that would be helpful in a class comment.